src/aks-preview/azext_aks_preview/_validators.py

```diff
@@ -587,6 +591,15 @@ def validate_azure_keyvault_kms_key_id(namespace):
             raise InvalidArgumentValueError(err_msg)
 
 
+def validate_azure_keyvault_kms_key_vault_resource_id(namespace):
+    key_vault_resource_id = namespace.azure_keyvault_kms_key_vault_resource_id
+    if key_vault_resource_id is None or key_vault_resource_id == '':
+        return
+    from msrestazure.tools import is_valid_resource_id
+    if not is_valid_resource_id(key_vault_resource_id):
+        raise CLIError("--azure-keyvault-kms-key-vault-resource-id is not a valid Azure resource ID.")
```

Comment:
Please use the specific error type `InvalidArgumentValueError` instead of `CLIError`

Reply:
thanks for pointing it out. fixed.